qutebrowser/browser/greasemonkey.py

```diff
@@ -325,7 +329,7 @@ def _add_script(self, script):
         log.greasemonkey.debug("Loaded script: {}".format(script.name))
 
     def _required_url_to_file_path(self, url):
-        requires_dir = os.path.join(_scripts_dir(), 'requires')
+        requires_dir = os.path.join(_scripts_dirs()[0], 'requires')
```

Comment:
This means if we download required files for a userscript in the config folder, that file will land in the data folder - not sure if that's a good idea? @toofar what do you think?

Reply:
I wouldn't be too worried either way, I don't think people have complained about those files so maybe they haven't needed to poke at them? Although the os.mkdir() below will fail if they have a config/greasemonkey but not a data/greasemonkey. So I think the options are:
1. add a source_path to GreasemonkeyScript and pass that through when constructing the download path so it can figure it out precisely
2. prefer the greasemonkey folder that exists (and just do the same as here if both exist) and assume anyone who has both can manage it well enough
3. just use os.makedirs below and always write to /data
4. move them to the cache folder (any migration required?)

I think I would prefer 2 over 3 to avoid creating an extra greasemonkey folder in a different place than the user created on. Of 4 if you think that makes sense and it doesn't end up being too much work to fit into this PR.

Reply:
We already do create both folders below in `init()` anyways, so I think this should work fine. IMHO it's not a problem if qutebrowser creates empty folders in its directories, since that also helps discoverability.